### PR TITLE
edit+delete sponsor

### LINKED ISSUE #89: Sponsor deleting
Створити можливість реалізації видалення профілю спонсора

### PR DESCRIPTION
**Pull request by daniilievtukhov**
closed #89 #88

### CODE REVIEW

File: src/components/SponsorPage/SponsorPage.jsx [hunk 1]
```diff
@@ -7,30 +7,42 @@ import { TalentsService } from "../../services/api-services";
...
+        if (editting) {
+            navigate(`${location.pathname}${location.search}#edit`, {
+                replace: true,
+            });
+        } else if (!editting) {
+            navigate(`${location.pathname}${location.search}`, {
+                replace: true,
+            });
+        }
+    }, [editting, navigate, location.pathname, location.search]);
```
Comment (Ruslanchik01) [inline src/components/SponsorPage/SponsorPage.jsx]: Why you write same thing 2nd time, look at rows 29-33

File: src/components/SponsorPage/SponsorPage.jsx [hunk 2]
```diff
@@ -77,32 +94,113 @@ export function SponsorPage() {
...
+                TalentsService.editSponsor(user.id, obj, token)
+                    .then(() => {
+                        setSaveError("");
+                        setProfile((prev) => ({
+                            ...prev,
+                            first_name: firstName.name,
+                            last_name: lastName.name,
+                            amount: numberKudos.kudos,
+                        }));
+                        setKudos(kudos + parseInt(numberKudos.kudos));
```
Comment (Ruslanchik01) [inline src/components/SponsorPage/SponsorPage.jsx]: use previous state to change this constant
```suggestion
                        setKudos((prev)=> prev + parseInt(numberKudos.kudos));
```

Comment (Ruslanchik01) [inline src/components/SponsorPage/SponsorPage.jsx]: use previous state to change this constant
```suggestion
                        setKudos((prev)=> prev + parseInt(numberKudos.kudos));
```

File: src/components/Layout/Header/Header.jsx [hunk 3]
```diff
@@ -35,8 +35,13 @@ export function Header() {
...
                     console.log(error);
                 });
+            TalentsService.getSponsorKudoses(user.id, token)
+                .then((kudos) => {
+                    setKudos(kudos.amount);
+                })
+                .catch((err) => console.log(err));
         }
-    }, [user.id]);
+    }, [user.id, kudos.amount]);
```
Comment (Ruslanchik01) [inline src/components/Layout/Header/Header.jsx]: There is no such key in kudos (state)

File: src/components/SponsorPage/SponsorPage.jsx [hunk 4]
```diff
@@ -45,29 +57,34 @@ export function SponsorPage() {
...
+    }, [profile]);
 
-    //         setSaveError("");
-    //     }
-    // }, [profile, editting]);
+    useEffect(() => {
+        if (Object.keys(profile).length !== 0) {
+            setFirstName({ name: profile?.first_name, error: "", state: true });
+            setLastName({ name: profile?.last_name, error: "", state: true });
+            setNumberKudos({ kudos: kudos?.amount, error: "", state: true });
```
Comment (Ruslanchik01) [inline src/components/SponsorPage/SponsorPage.jsx]: There is no such kudos.amount

File: src/components/SponsorPage/components/SponsorAbout/SponsorAbout.jsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+} from "react";
+import { UserContext } from "../../../../context/UserContext";
+import { Button, Textarea } from "../../../../shared/components";
+import { TalentsService } from "../../../../services/api-services";
+import edit from "./images/edit.svg";
+import s from "./SponsorAbout.module.scss";
+
+export const SponsorAbout = forwardRef((props, ref) => {
+    const {
+        profile,
```
Comment (Ruslanchik01) [inline src/components/SponsorPage/components/SponsorAbout/SponsorAbout.jsx]: Don't use this prop, so no sense to transfer it

File: src/components/SponsorPage/components/SponsorData/SponsorData.jsx [hunk 6]
```diff
@@ -1,36 +1,147 @@
...
-                .then((kudos) => {
-                    setKudos(kudos.amount);
-                })
-                .catch((err) => console.log(err));
-        }
-    }, [user.id]);
+export const SponsorData = forwardRef((props, ref) => {
+    const { user, token, kudos, setKudos, userInfo } = useContext(UserContext);
+    const {
+        profile,
```
Comment (Ruslanchik01) [inline src/components/SponsorPage/components/SponsorData/SponsorData.jsx]: You don't transfer this prop in parent component and don't use in this

File: src/components/SponsorPage/SponsorPage.jsx [hunk 7]
```diff
@@ -77,32 +94,113 @@ export function SponsorPage() {
...
+                .catch((error) => {
+                    console.error(error);
+                });
+        } catch (error) {
+            console.error(error);
+        }
+    };
+
+    const propsTalentData = {
+        user,
```
Comment (Ruslanchik01) [inline src/components/SponsorPage/SponsorPage.jsx]: Why you transfer it, if it won't be used

File: src/components/SponsorPage/SponsorPage.jsx [hunk 8]
```diff
@@ -77,32 +94,113 @@ export function SponsorPage() {
...
+        editting,
+        firstName,
+        setFirstName,
+        lastName,
+        setLastName,
+        numberKudos,
+        setNumberKudos,
+        state,
```
Comment (Ruslanchik01) [inline src/components/SponsorPage/SponsorPage.jsx]: same problem

File: src/components/SponsorPage/SponsorPage.jsx [hunk 9]
```diff
@@ -7,30 +7,42 @@ import { TalentsService } from "../../services/api-services";
...
-    // const { setUserInfo } = useContext(UserContext);
-    // const [modalIsOpen, setModalIsOpen] = useState(false);
-    // const [cancelModalIsOpen, setCancelModalIsOpen] = useState(false);
-    // const [saveError, setSaveError] = useState("");
-
+    const { setUserInfo, kudos, setKudos } = useContext(UserContext);
+    const [modalIsOpen, setModalIsOpen] = useState(false);
+    const [cancelModalIsOpen, setCancelModalIsOpen] = useState(false);
+    const [saveError, setSaveError] = useState("");
+    const [state, setState] = useState(0);
```
Comment (Ruslanchik01) [inline src/components/SponsorPage/SponsorPage.jsx]: useless state, is never used

File: src/components/SponsorPage/SponsorPage.jsx [hunk 10]
```diff
@@ -77,32 +94,113 @@ export function SponsorPage() {
...
-    // };
+            try {
+                TalentsService.editSponsor(user.id, obj, token)
+                    .then(() => {
+                        setSaveError("");
+                        setProfile((prev) => ({
+                            ...prev,
+                            first_name: firstName.name,
+                            last_name: lastName.name,
+                            amount: numberKudos.kudos,
```
Comment (Ruslanchik01) [inline src/components/SponsorPage/SponsorPage.jsx]: You never use this value of profile

File: src/components/SponsorPage/components/SponsorAbout/SponsorAbout.jsx [hunk 11]
```diff
@@ -0,0 +1,63 @@
+import React, {
+    useCallback,
+    forwardRef,
+    useImperativeHandle,
```
Comment (Ruslanchik01) [inline src/components/SponsorPage/components/SponsorAbout/SponsorAbout.jsx]: Unused import

File: src/components/SponsorPage/components/SponsorAbout/SponsorAbout.jsx [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+    useState,
+    useContext,
+} from "react";
+import { UserContext } from "../../../../context/UserContext";
+import { Button, Textarea } from "../../../../shared/components";
+import { TalentsService } from "../../../../services/api-services";
+import edit from "./images/edit.svg";
+import s from "./SponsorAbout.module.scss";
+
+export const SponsorAbout = forwardRef((props, ref) => {
```
Comment (Ruslanchik01) [inline src/components/SponsorPage/components/SponsorAbout/SponsorAbout.jsx]: No sense to use forwardRef

**Review by Ruslanchik01 — CHANGES_REQUESTED** [review]

**Review by whoisalex1W — APPROVED** [review]

File: src/components/SponsorPage/SponsorPage.jsx (hunk 4 above)
Comment (Ruslanchik01) [reply src/components/SponsorPage/SponsorPage.jsx]: > There is no such kudos.amount

dont resolved

File: src/components/SponsorPage/SponsorPage.jsx (hunk 10 above)
Comment (Ruslanchik01) [reply src/components/SponsorPage/SponsorPage.jsx]: > You never use this value of profile

don't resolved

File: src/components/SponsorPage/components/SponsorAbout/SponsorAbout.jsx [hunk 13]
```diff
@@ -0,0 +1,53 @@
+import React, { forwardRef } from "react";
```
Comment (Ruslanchik01) [inline src/components/SponsorPage/components/SponsorAbout/SponsorAbout.jsx]: delete forwardRef

**Review by Ruslanchik01 — CHANGES_REQUESTED** [review]

**Pull request merged**